entity-data-service-rx-client/src/main/java/org/hypertrace/entity/data/service/rxclient/EntityDataCachingClient.java

```diff
@@ -121,8 +120,8 @@
     private UpsertCondition condition;
 
     private void executeUpdate() {
-      // Acquire write lock to ensure no more modification of this update
-      Lock lock = pendingUpdateStripedLock.get(entityKey).writeLock();
+      // Make sure no current additions
+      Lock lock = pendingUpdateStripedLock.get(entityKey);
```

Comment:
L133 seems redundant since cache update is already done within L132. Can be removed.

Reply:
two different uses of the word cache.
```
Single<Entity> updateResult =
          EntityDataCachingClient.this.createOrUpdateEntity(entityKey, condition).cache();
Caches in the Single - that is, all subscribers will share a result rather than re-executing 
```
( http://reactivex.io/RxJava/3.x/javadoc/io/reactivex/rxjava3/core/Single.html#cache-- )
```
EntityDataCachingClient.this.cache.put(entityKey, updateResult);
```
puts the single itself into the guava cache for future callers

Reply:
Yes, but if we go inside:
```
private Single<Entity> createOrUpdateEntity(
      EntityKey entityKey, UpsertCondition upsertCondition) {
    Single<Entity> updateResult =
        this.upsertEntityWithoutCaching(entityKey, upsertCondition).cache();
    EntityDataCachingClient.this.cache.put(entityKey, updateResult);
    return updateResult;
  }
```
that is already adding the single within the guava cache, right?

Reply:
Thanks for following up on this, my mind is clearly all over the place 🙁 - addressed.